cmd/auth/internal/application/eventhandler/register.go

```diff
@@ -25,10 +25,16 @@ func Register(conn *grpc.ClientConn, eventBus eventbus.EventBus, topicToHandlerM
 
 			for topic, handler := range topicToHandlerMap {
 				// Will resubscribe to handler on error infinitely
-				go gollback.New(context.Background()).Retry(0, func(ctx context.Context) (interface{}, error) {
-					err := eventBus.Subscribe(ctx, topic, handler)
-					return nil, errors.Newf(errors.INTERNAL, "EventHandler %s unsubscribed (%v)", topic, err)
-				})
+				go func(topic string, handler eventbus.EventHandler) {
```

Comment:
Instead of wrapping it with a goroutine simply change 0 to some specific value, for example 5

https://github.com/vardius/gollback/blob/master/README.md

‘’’
// Will retry 5 times or will timeout by context
	res, e := g.Retry(5, func(ctx context.Context) (interface{}, error) {
		return nil, errors.New("failed")
	})
‘’’

where you can handle error